File: bms/go/triangolatte/calculateNormals/calculateNormals.go
package test

import (
	"math"
)

var ()

const ()

func calculateNormals(x, y float64) [2]Point {
	return [2]Point{
		Point{y, -x}.Normalize(),
		Point{-y, x}.Normalize(),
	}
}

func (p Point) Normalize() Point {
	return p.Scale(1)
}

func (p Point) Scale(f float64) Point {
	norm := float64(math.Sqrt(float64(p.X*p.X + p.Y*p.Y)))
	return Point{p.X / norm * f, p.Y / norm * f}
}

type Point struct {
	X, Y float64
}

File: translations_only/fail/go/triangolatte/calculateNormals/gpt4/Hinted/calculateNormals.rs

use std::f64;

struct Point {
    x: f64,
    y: f64,
}

fn calculate_normals(x: f64, y: f64) -> [Box<Point>; 2] {
    [
        Box::new(normalize(Point { x: y, y: -x })),
        Box::new(normalize(Point { x: -y, y: x })),
    ]
}

fn normalize(p: Point) -> Point {
    scale(p, 1.0)
}

fn scale(p: Point, f: f64) -> Point {
    let norm = f64::sqrt(p.x * p.x + p.y * p.y);
    Point {
        x: p.x / norm * f,
        y: p.y / norm * f,
    }
}

fn main() {}
